### PR TITLE
help: rewrap output to 80 columns

### PR DESCRIPTION
**Pull request by EricFromCanada**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [n/a] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
Adds a call to `Formatter.wrap` to CLI::Parser so that descriptions of commands are pre-wrapped when running `brew help <command>`. Also added to `help.rb` for for commands still using comment-based descriptions. 

`Formatter.wrap` now does the following:
- converts any hard-wrapped paragraphs to a single line
- wraps any option descriptions with an indent that matches what's automatically generated by CLI::Parser (only found in comment-based descriptions) and then rewraps any wrapped overflowing indented lines to the given width
- wraps each non-indented line to the given width

Also fixes help text in `--cellar`, `analytics`, `install`, `list`, `prof`, `search`, `unlink`, and `vendor-install`.

### CODE REVIEW

File: Library/Homebrew/utils/formatter.rb [hunk 1]
```diff
@@ -32,7 +32,12 @@ def error(string, label: nil)
   end
 
   def wrap(s, width = 172)
-    s.gsub(/(.{1,#{width}})(\s+|\Z)/, "\\1\n")
+    col = 43
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/formatter.rb]: why 43?

File: Library/Homebrew/utils/formatter.rb [hunk 2]
```diff
@@ -32,7 +32,12 @@ def error(string, label: nil)
...
+    indent = width - col
+    s.gsub(/(?<=\S) *\n(?=\S)/, " ")
+     .gsub(/^( +-.+  +(?=\S.{#{col}}))(.{1,#{col}})( +|$)\n?/, "\\1\\2\n" + " " * indent)
+     .gsub(/^( {#{indent}}(?=\S.{#{col}}))(.{1,#{col}})( +|$)\n?/, "\\1\\2\n" + " " * indent)
+     .gsub(/(.{1,#{width}})( +|$)\n?/, "\\1\n")
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/formatter.rb]: Could you add a bunch more comments here? At a glance it's very hard to figure out what's going on here. Also: does this wrap at whitespace or in the middle of words?

File: Library/Homebrew/utils/formatter.rb (hunk 1 above)
Comment (EricFromCanada) [reply Library/Homebrew/utils/formatter.rb]: This matches the value of `@desc_line_length` in `cli_parser.rb`, used when it wraps option descriptions for parser-based commands.

File: Library/Homebrew/utils/formatter.rb (hunk 2 above)
Comment (EricFromCanada) [reply Library/Homebrew/utils/formatter.rb]: Yes, regular expressions can be like that. 🙃

The first line turns any hard-wrapped paragraphs into a single line. The other three are different takes on the same concept: find a run of characters up to length `n`, then backtrack  (unless the line is already shorter than `n`) until a space is found and replace it with a line break. The algorithm is lifted from this [blog post](https://macromates.com/blog/2006/wrapping-text-with-regular-expressions/), except it doesn't attempt to also line-break words longer than `n`, e.g. URLs.

File: Library/Homebrew/utils/formatter.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/utils/formatter.rb]: Can you pull both into a constant? Thanks!

File: Library/Homebrew/utils/formatter.rb (hunk 2 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/utils/formatter.rb]: Great, thanks. Would be good to document your comments here inline in Ruby and link the blog post too?

**Comment by MikeMcQuaid** [comment]
Great, thanks again @EricFromCanada!

**Pull request merged**